jasmine-standalone-3.6.0/src/BowlingGame.js

```diff
@@ -0,0 +1,105 @@
+class BowlingGame {
+
+  constructor() {
+    this.frameRolls = [];
+    this.frameScores = [];
+    this.STRIKE_VALUE = 10;
+    this.INDEX_OF_FINAL_FRAME = 9;
+  }
+
+  playFrame(roll1, roll2 = 'no throw') {
+    if (this.frameRolls.length >= 10) {
+      throw new Error(`You have already played ${this.INDEX_OF_FINAL_FRAME + 1} frames`)
+    }
+    this._addRolls(roll1, roll2);
+    this._addBonusScores();
+  }
+
+  getGameScore() {
+    return this.frameScores.reduce(function(a,b) {
+      return a + b;
+    }, 0);
+  }
+
+  areBonusRollsNeeded() {
+    if (this._getFrameTotal(this.frameRolls[this.INDEX_OF_FINAL_FRAME]) === this.STRIKE_VALUE && this.frameRolls[this.INDEX_OF_FINAL_FRAME].length === 2) {
```

Comment:
You're right in that one line boolean methods are a way of tidying this up, even if they're only called once. The long lines of code can be a little tricky to read.

Reply:
Also used in tandem with ternary operators: https://www.javascripttutorial.net/javascript-ternary-operator/